apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -334,14 +383,19 @@ public SpannerConnectionStringBuilder(string connectionString, ChannelCredential
         public SpannerConnectionStringBuilder(string connectionString, ChannelCredentials credentials, SessionPoolManager sessionPoolManager)
         {
             ConnectionString = GaxPreconditions.CheckNotNull(connectionString, nameof(connectionString));
-            CredentialOverride = credentials;
+            // If SPANNER_EMULATOR_HOST env var is set, ignore any provided credentials and use insecure credentials.
+            ParseEmulatorHostAndPort();
```

Comment:
You parse here, but also you keep parsing whenever the Host or the Port are requested, so it might be that at the time of construction the emulator was not set, but that it is set when Host or Port are requested, and then the credentials would be wrong.
I would only parse on construction.

Reply:
No longer using this logic.